tests/model_monitoring/test_tdengine.py

```diff
@@ -237,3 +238,127 @@ def test_get_records_query(
+                datetime.datetime.now().astimezone() - datetime.timedelta(hours=2),
+                datetime.datetime.now().astimezone() - datetime.timedelta(hours=1),
+                "time_column_v2",
+                "3h",
+                50,
+                ["avg", "max", "sum"],
+                "12m",
+            ),
+        ],
+    )
+    def test_get_records_with_interval_query(
+        self,
+        super_table: TDEngineSchema,
+        subtable: str,
+        columns_to_filter: list[str],
+        start: str,
+        end: str,
+        timestamp_column: str,
+        interval: str,
+        limit: int,
+        agg: list,
+        sliding_window: str,
+    ):
+        with pytest.raises(mlrun.errors.MLRunInvalidArgumentError) as err:
+            # Provide aggregation functions without columns to filter
+            super_table._get_records_query(
+                table=subtable,
+                start=start,
+                end=end,
+                timestamp_column=timestamp_column,
+                interval=interval,
+                limit=limit,
+                agg=agg,
+                sliding_window=sliding_window,
+            )
+            assert (
+                "columns_to_filter must be provided when using aggregate functions"
+                in str(err.value)
+            )
+
+        with pytest.raises(mlrun.errors.MLRunInvalidArgumentError) as err:
+            # Provide aggregation functions without interval
+            super_table._get_records_query(
+                table=subtable,
+                start=start,
+                end=end,
+                columns_to_filter=columns_to_filter,
+                timestamp_column=timestamp_column,
+                limit=limit,
+                agg=agg,
+                sliding_window=sliding_window,
+            )
+            assert (
+                "both interval and aggregate function must be provided or neither"
+                in str(err.value)
+            )
+
+        with pytest.raises(mlrun.errors.MLRunInvalidArgumentError) as err:
+            # Provide sliding window without interval
+            super_table._get_records_query(
+                table=subtable,
+                start=start,
+                end=end,
+                columns_to_filter=columns_to_filter,
+                timestamp_column=timestamp_column,
+                limit=limit,
+                agg=agg,
+                sliding_window=sliding_window,
+            )
+            assert "interval must be provided when using sliding window" in str(
+                err.value
+            )
+        columns_to_select = ", ".join(
+            [f"{a}({col})" for a in agg for col in columns_to_filter]
+        )
+        expected_query = (
+            f""
+            f"SELECT _wstart, _wend, {columns_to_select} from {_MODEL_MONITORING_DATABASE}.{subtable} "
+            f"where {timestamp_column} >= '{start}' and {timestamp_column} <= '{end}' "
```

Comment:
Not sure why `SELECT` and `LIMIT` are uppercase but `from`, `where` and `and` are lowercase? Better keep them consistent.